### PR TITLE
Chore/activeadmin upgrade

### PR DESCRIPTION
**Pull request by tsubik**
Upgrading ActiveAdmin to v3 as well as upgrading all gems.

### CODE REVIEW

File: app/models/api_key.rb [hunk 1]
```diff
@@ -21,4 +21,8 @@ class APIKey < ApplicationRecord
   def expired?
     DateTime.now >= expires_at || user.deactivated? || deactivated?
   end
+
+  def self.ransackable_attributes(auth_object = nil)
```
Comment (santostiago) [inline app/models/api_key.rb:25]: mmm... what is this for?

File: app/models/fmu.rb [hunk 2]
```diff
@@ -67,7 +67,7 @@ class Fmu < ApplicationRecord
   scope :filter_by_operators, ->(operator_ids) { joins(:fmu_operators).where(fmu_operators: {current: true, operator_id: operator_ids.split(",")}) }
   # this could also be done like: "id not in ( select fmu_id from fmu_operators where \"current\" = true)"
   # but it might break the method chaining
-  scope :filter_by_free, -> { where.not(id: FmuOperator.where(current: true).pluck(:fmu_id)).group(:id) }
+  scope :filter_by_free, -> { where.not(id: FmuOperator.where(current: true).select(:fmu_id)).group(:id) }
```
Comment (santostiago) [inline app/models/fmu.rb:70]: 👍

File: config/initializers/active_record.rb [hunk 3]
```diff
@@ -8,4 +8,16 @@ def structurally_incompatible_values_for_or(other)
         (Relation::CLAUSE_METHODS - [:having, :where]).reject { |m| send("#{m}_clause") == other.send("#{m}_clause") }
     end
   end
+
+  # After upgrading to ransack 4.0 we need to explicitly whitelist attributes
+  # In this project I prefer to only blacklist the attributes that are not allowed for couple models
+  class Base
```
Comment (santostiago) [inline config/initializers/active_record.rb:14]: oh...

File: db/migrate/20210607152721_add_current_unique_index_to_score_operator_documents.rb [hunk 4]
```diff
@@ -1,3 +1,4 @@
+# rubocop:disable all
```
Comment (santostiago) [inline db/migrate/20210607152721_add_current_unique_index_to_score_operator_documents.rb:1]: We could just add the folder migrations to the Rubocop exceptions.

File: lib/file_data_import/belongs_to_association.rb [hunk 5]
```diff
@@ -62,7 +62,7 @@ def translations_attributes
 
     def extract_attributes(attrs)
       attrs.each_with_object({}) do |attribute, attributes|
-        value = raw_attributes["#{singular_name}__#{attribute}".to_sym]
+        value = raw_attributes[:"#{singular_name}__#{attribute}"]
```
Comment (santostiago) [inline lib/file_data_import/belongs_to_association.rb:65]: this also changed?

**Review by santostiago — APPROVED** [review]
👍

File: app/models/api_key.rb (hunk 1 above)
Comment (tsubik) [reply app/models/api_key.rb:25]: To blacklist all attributes of this model to not be usable in ransack queries.

File: db/migrate/20210607152721_add_current_unique_index_to_score_operator_documents.rb (hunk 4 above)
Comment (tsubik) [reply db/migrate/20210607152721_add_current_unique_index_to_score_operator_documents.rb:1]: Yeah, that could be another option, but I thought we might need rubocop for new migrations. It has a lot of very useful warnings and hints to write better migrations.

File: lib/file_data_import/belongs_to_association.rb (hunk 5 above)
Comment (tsubik) [reply lib/file_data_import/belongs_to_association.rb:65]: One of rubocop auto corrections.

**Pull request merged**